components/activity_and_insights/insights/top_boards/top_boards.tsx

```diff
@@ -0,0 +1,116 @@
+import widgetHoc, {WidgetHocProps} from '../widget_hoc/widget_hoc';
+import WidgetEmptyState from '../widget_empty_state/widget_empty_state';
+
+import './../../activity_and_insights.scss';
+
+const TopBoards = (props: WidgetHocProps) => {
+    const [loading, setLoading] = useState(true);
+    const [topBoards, setTopBoards] = useState([] as TopBoard[]);
+
+    const currentTeamId = useSelector(getCurrentTeamId);
+    const boardsHandler = useSelector((state: GlobalState) => state.plugins.insightsHandlers.focalboard);
+
+    const getTopBoards = useCallback(async () => {
+        setLoading(true);
+        const data: any = await boardsHandler(props.timeFrame, 0, 4, currentTeamId, props.filterType);
+        if (data.items) {
+            setTopBoards(data.items);
+        }
+        setLoading(false);
+    }, [props.timeFrame, currentTeamId, props.filterType]);
+
+    useEffect(() => {
+        getTopBoards();
+    }, [getTopBoards]);
+
+    const skeletonLoader = useCallback(() => {
+        const entries = [];
+        for (let i = 0; i < 4; i++) {
+            entries.push(
+                <div
+                    className='top-board-loading-container'
+                    key={i}
+                >
+                    <CircleLoader
+                        size={32}
+                    />
+                    <div className='loading-lines'>
+                        <TitleLoader/>
+                        <TitleLoader/>
+                    </div>
+                </div>,
+            );
+        }
+        return entries;
+    }, []);
+
+    return (
+        <div className='top-board-container'>
+            {
+                loading &&
+                skeletonLoader()
+            }
+            {
+                (topBoards && !loading) &&
+                <div className='board-list'>
+                    {
+                        topBoards.map((board, i) => {
+                            return (
+                                <div
+                                    className='board-item'
+                                    onClick={() => {}}
+                                    key={i}
+                                >
+                                    <span className='board-icon'>{board.icon}</span>
+                                    <div className='display-info'>
+                                        <span className='display-name'>{board.title}</span>
+                                        <span className='update-counts'>
+                                            <FormattedMessage
+                                                id='insights.topBoards.updates'
+                                                defaultMessage='{updateCount} updates'
+                                                values={{
+                                                    updateCount: board.activityCount,
+                                                }}
```

Comment:
Will the object passed to the `values` property be recreated each time causing unnecessary renders?

Reply:
I took a look at the profiler and if I use a function with useCallback it's the same number of renders because this component only re-renders on prop change anyway